linkerd/server-policy/src/lib.rs

```diff
@@ -2,17 +2,25 @@ mod network;
 
 pub use self::network::Network;
 use std::{
-    collections::{HashMap, HashSet},
+    collections::{BTreeMap, HashSet},
+    hash::Hash,
+    iter::FromIterator,
+    ops::Deref,
+    sync::Arc,
     time,
 };
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct ServerPolicy {
     pub protocol: Protocol,
     pub authorizations: Vec<Authorization>,
-    pub labels: HashMap<String, String>,
+    pub labels: Labels,
 }
 
+/// Stores an ordered, cloneeable set of labels.
```

Comment:
nit: i don't _think_ there should be two 'e's here :)
```suggestion
/// Stores an ordered, cloneable set of labels.
```